### PR TITLE
update grails-shell to v7.0.0-SNAPSHOT

### PR DESCRIPTION
**Pull request by jamesfredley**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -74,10 +74,20 @@ java {
     withSourcesJar()
 }
 
-configurations.configureEach {
-    // This is a workaround for grails-shell exposing a different version of Groovy
-    // than the one used by Gradle. This causes issues with the Groovy compiler.
-    resolutionStrategy.force "org.codehaus.groovy:groovy-xml:$GroovySystem.version"
+configurations.all {
```
Comment (matrei) [inline build.gradle]: Why do you choose `configurations.all` instead of `configurations.configureEach`?

File: build.gradle [hunk 2]
```diff
@@ -74,10 +74,20 @@ java {
...
+    resolutionStrategy.eachDependency { DependencyResolveDetails details ->
+        // This is a workaround for grails-shell exposing a different version of Groovy
+        // than the one used by Gradle. This causes issues with the Groovy compiler.
+        if (details.requested.group == 'org.apache.groovy') {
+            details.useTarget(group: "org.codehaus.groovy", name: details.requested.name, version: "$GroovySystem.version")
```
Comment (matrei) [inline build.gradle]: GroovySystem.version returns a String, no need to use GString.

File: build.gradle [hunk 3]
```diff
@@ -74,10 +74,20 @@ java {
...
+            details.because "Use Groovy 3.0.x library shipped with Gradle"
+        }
+
+        if (details.requested.group == 'org.codehaus.groovy') {
+            details.useTarget(group: "org.codehaus.groovy", name: details.requested.name, version: "$GroovySystem.version")
```
Comment (matrei) [inline build.gradle]: GroovySystem.version returns a String, no need to use GString.

**Review by matrei — APPROVED** [review]

File: build.gradle (hunk 1 above)
Comment (jamesfredley) [reply build.gradle]: It was just copied from another Grails project, I have adjusted it to configurations.configureEach which works well.

File: build.gradle (hunk 2 above)
Comment (jamesfredley) [reply build.gradle]: Updated

File: build.gradle (hunk 3 above)
Comment (jamesfredley) [reply build.gradle]: Updated

**Pull request merged**